### PR TITLE
Phase0 HE in 2017, Phase1 HE in 2018

### PR DESCRIPTION
**Pull request by kpedro88**
This PR reverts the 2017 HE back to the Phase0 model (HPDs w/ 2-3 depth segmentations). It supersedes #17052. It was developed on top of `CMSSW_9_0_X_2017-01-15-2300`.

856a5b08697fec72112a7e9ec5ec2bc6b05b4336 applies several TEMPORARY changes to make the 2017 workflow continue to run:
* customise function to use hardcode conditions for HCAL and skip packing/unpacking (due to lack of appropriate conditions for this geometry, including emap)
* use of XML geometry instead of database (updated 2017 geom not yet in DB tag)

Other notable changes in this PR:
* HBHEPhase1Reconstructor will still be used for 2017, even though we only plan to have HPDs - better/more maintainable code, already validated in DPG
* The "TestNumbering" setting is now enabled for 2017 onward - this allows the reuse of GEN-SIM when HB or HE topology (=depth segmentation) is changed, by storing SimHits per-layer instead of per-depth
* 2018 workflow introduced to keep the Phase1 HE alive for local HCAL studies (didn't bother to duplicate all the extra cosmic/tracker/etc. workflow variations, they can be added if someone needs them)

My intention is that @ianna and @abdoulline + AlCa contacts will make new tags for changed HCAL conditions and geometry records (for *both* 2017 and 2018 = clone of current 2017), and then I can update `autoCond.py` in this PR for final tests. Therefore, I will put the PR on hold once submitted so it will not be merged without these necessary pieces.

This PR will be backported to 81X, but #16874 needs to be merged first or HCAL validation will not be able to handle the TestNumbering SimHits.

### CODE REVIEW

**Comment by kpedro88** [comment]
hold

**Comment by kpedro88** [comment]
please test

**Comment by ianna** [comment]
@kpedro88 and @mmusich - the new tags based on this PR are in DB:
```
XMLFILE_Geometry_90YV2_Extended2017_mc
HCALParameters_Geometry_90YV2
HCALRECO_Geometry_90YV2
```

**Comment by kpedro88** [comment]
@ianna thanks! Should we just reuse the old tags for the 2018 geometry (corresponds to the previous incarnation of 2017, including the upgraded HE), or would it be better to clone them to have 2018 in the name?

**Comment by ianna** [comment]
@kpedro88 - I do not think we can reuse tags with different labels if it's the same GT queue. Besides, neither HCal reco geometry nor HCal parameter tags have labels. However, you can always use a specific GT, a.k.a. the "old" one. 

In any case, it is not the last modification - there is a GEM slice test and Tracker&Pixel overlap fixes to be included (in pending PRs)

**Comment by kpedro88** [comment]
My goal would be to make a separate GT queue for 2018, so we can keep alive all the work already done to prepare MC conditions for the upgraded HE. But I guess AlCa has to create this queue before you can make tags for it.

**Comment by ianna** [comment]
@kpedro88 - if it's a separate GT queue why not take the existing tags? I would not thing having GEM in or out makes much difference for your studies. 

@mmusich - would it be possible to clone existing upgrade2017 as upgrade2018 before updating existing geometry tags?

**Comment by mmusich** [comment]
@ianna about:
 > would it be possible to clone existing upgrade2017 as upgrade2018 before updating existing geometry tags?

it's certainly is, I let @arunhep and @franzoni to decide what is the best course of action in this respect.

**Comment by franzoni** [comment]
AlCa is looking at this PR.
@mmusich is honorable :-) former convenor, as he reported again.
We'll take actions and feedback in due time, soon.

**Comment by ianna** [comment]
@franzoni - thanks!

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -1701,6 +1701,8 @@ def lhegensim(fragment,howMuch):
 defaultDataSets={}
 defaultDataSets['2017']='CMSSW_8_1_0_pre16-81X_upgrade2017_realistic_v22-v'
 defaultDataSets['2017Design']='CMSSW_8_1_0_pre16-81X_upgrade2017_design_IdealBS_v6-v'
+defaultDataSets['2018']='CMSSW_8_1_0_pre16-81X_upgrade2017_realistic_v22-v'
+defaultDataSets['2018Design']='CMSSW_8_1_0_pre16-81X_upgrade2017_design_IdealBS_v6-v'
```
Comment (franzoni) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: These GEN-SIM were made with the phase1 HE
and won't be consistent with the re-defined 2017 scenario

Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: Yes, they'll be appropriate for the new 2018, but not the new 2017. We don't have any alternatives at the moment, though. It should be updated once we have new relvals.

**Comment by franzoni** [comment]
hello @kpedro88 

the re-definition of the 2017 scenario as you propose in this PR is the "HE back to the Phase0 model". This is also what TSG was advocating yesterday (ORP Jan 17, 2017) for their immediate needs of samples production.

For 2017 do we need / plan to have to support both
(in terms or cmsDriver era, geometry, digi and reco conditions, etc):
. "HE back to the Phase0 model"
. as well as "HE back to the Phase0 except 1 wedge instrumented ala phase1-2018"
This was not clear at the discussion which took place here https://indico.cern.ch/event/599077/ 

I this needs input, besides HCAL DPG, also from all stakeholders
@davidlange6

**Comment by kpedro88** [comment]
@franzoni The "2018" scenario in this PR is not the "1 wedge upgraded" case. It is exactly the current 2017 scenario including the *entire* upgraded HE.

The 2018 components could become a separate PR, but I am reluctant to allow any interruption of support for a scenario which we *know* works and which is actively being used for MC studies.

Because the 2018 scenario is just a clone of the current 2017 scenario (w/ appropriate geometry and Era already present in this PR), my impression was that it would not be too much additional work to copy the current state of the 2017 GT queue to make a 2018 queue. If this is not the case, then we can postpone it to a separate PR, but again, interruptions of support should be minimized.

**Comment by franzoni** [comment]
> interruptions of support should be minimized

agreed.

My question was different, though. Rephrasing: for the 2017 HCAL, 
will there be only the scenario you put forward in this PR (i.e. HE rolled back to phase0)
or will there be also an additional one with the "HE back to the Phase0 except 1 wedge instrumented ala phase1-2018" ?
In your nomenclature, the full HE-2018 is the complete phase1 HCAL.

**Comment by kpedro88** [comment]
I personally oppose the 1-wedge installation and think it should not be simulated (even if some hacks are needed for data reco). Nevertheless the final decision has not been made yet (and may not be made until a month from now), so I can't offer a definitive answer.

**Comment by abdoulline** [comment]
I gree with Kevin, let's now proceed with "Plan 0" for 2017 (only HF 
Upgrade) while "Plan 1" (Upgrade of HF + 1 out of 36 HE RBXs) is 
under consideration(s)...

**Comment by franzoni** [comment]
We were expecting the workflow 10024 (2017 ttbar) only to have HCAL and derived objects -related changes. Instead, at least based on the 10 events test, differences arise also in tracker-tracking related quantities, see: https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2017-01-17-1100+17198/17903/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/

Is this understood ? Can we understand this as stemming from the changes in the two xml of hcal geometry ?


We're in parallel making queues and will provide commits to be cherry-picked.
ETA 24h.

**Comment by VinInn** [comment]
@franzoni could you please circulate the DQM with both samples : following the link on the baseLineComparison pahe I got only the new one
https://goo.gl/EoC37j

**Comment by makortel** [comment]
The tracking differences start already in the simulation
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2017-01-17-1100+17198/17903/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/Tracking_Track_simulation.html
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2017-01-17-1100+17198/17903/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/Tracking_TrackingMCTruth_TrackingParticle.html

If the 2017 SIM geometry is changed (I guess it is but am not sure), the G4 random number sequences will become different so an exact match can no longer be expected (anywhere).

**Comment by franzoni** [comment]
@vininn there is no dqm
We only have the plots made by the ib infrastructure, for the moment

Thanks @makortel
What mechanism is used to reproduce the random numbers ?

**Comment by makortel** [comment]
@franzoni What do you mean? The G4 generates the random numbers with generators from RandomNumberGeneratorService. The sequence of numbers depends on the history, so in principle even small change anywhere can eventually lead to noticeable differences when one looks only a single (or handful) of event(s).

**Comment by kpedro88** [comment]
Yes, @makortel is correct - any geometry change upsets the whole pseudorandom number pattern. Since we are now using "TestNumbering" for HCAL to save per-layer info for SimHits, this will definitely change the outcome of any GEN-SIM run.

In addition, the workflow is currently in a temporary state with packing/unpacking disabled due to the use of fake conditions. This can cause some changes or missing collections. Further, removing the SiPM simulation for HE will cause more random number changes in the DIGI simulation, as the SiPM simulation contains numerous random number calls.

I would not be too concerned about the comparison yet; I ran tests primarily to make sure the code still runs. Eventually we will have a more accurate comparison when the MC conditions are finalized. Still, it is best to be aware of the random number changes.

**Comment by franzoni** [comment]
hello @kpedro88 @abdoulline 

Please find this branch: 
https://github.com/kpedro88/cmssw/compare/Phase1-HE14...franzoni:gf_Phase1-HE14_from-CMSSW_9_0_X_2017-01-15-2300?expand=1
from which  @arunhep and I invite you to cherry pick commits into this PR.

Summary:

+ new GT's for 2018 have been associated to the phase1_2018_*  keys in autocond; they keep the same content as the past 2017 ones, they're now only consistently named.
  Added third key for cosmics

+ new 2017 GT's updated to v2: the change with respect to v1 are the 3 updated geometry tags (thanks @ianna) implementing the HE-phase0 roll back on top of the rest of the phase1 CMS
  The geometry for 2017 has been labelled Extended in the GT, as customary

+ the geometry for 2017 relvals has been put back to DB:Extended from database, since updating the GT's phase1_2017_* and having used the Extended label

+ the geometry for the new 2018 relvals has been put back to DB:Extended from database

Side nodes:
+ we gather the 2018 relvals actually need to be added to the matrix
+ understand from @abdoulline that the other conditions for 2017 HCAL digitization will come ~next week
+ we've created 2018 queues and will from now use the existing 2017 ones for the phase1-CMS-HE-rollback

**Comment by kpedro88** [comment]
@franzoni thanks, I added the commits.

There are already 2018 relvals in the matrix, see updates to relval_2017.py and runTheMatrix.py in this PR.

**Comment by kpedro88** [comment]
@franzoni also, we'll have to repeat this process once the 81X backport is submitted, but that is waiting on @davidlange6 to merge #16874

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by franzoni** [comment]
> we'll have to repeat this process once the 81X backport is submitted

sure. Shouldn't we we have the rest of the HCAL digitization-relevant db tags,
(and revert to using digis-from-raw, as opposed to sim-digis as now)
before back-porting to 81x

**Comment by abdoulline** [comment]
we're working on "Plan 0" DB conditions...

**Comment by slava77** [comment]
@kpedro88 
please clarify on  rather systemaitc removal of phase2_hcal.toModify.
Why is it not necessary anymore?

In a few places the removed phase2_hcal.toModify are replaced by run2_HCAL_2017.toModify which, IIUC, are temporary. The Phase2C2 era contains both phase2_hcal and run2_HCAL_2017 and it looks like there are no changes in phase2 outputs; but then if run2_HCAL_2017  edits are temporary, would there be need to revert back to phase2_hcal after they are cleaned up?

**Comment by kpedro88** [comment]
@slava77 run2_HCAL_2017 is not temporary. The TestNumbering settings will be used *from now on*, i.e. for 2017 configs, 2018, etc. including 2023. Since the Phase2 Era includes the 2017 HCAL customizations, the TestNumbering settings do not need to be duplicated in phase2_hcal.

**Comment by slava77** [comment]
@kpedro88 
is it still too early to pay attention to the changes in performance in this PR?
It looks like a bunch of condition payloads are to be prepared.
Will you remove the HCalCustoms.customise_Hcal2017 from the matrix workflow definitions after the GT is updated?

**Comment by kpedro88** [comment]
@slava77 yes, the comparisons still aren't useful yet. I will remove the customize once we have appropriate HCAL tags in the GT.

**Comment by franzoni** [comment]
hello @kpedro88 @abdoulline 

do you have an expectation of when the digitization-related tags may become available ?

@ianna gave today a comprehensive status report on what's coming from the geometry side ( see useful  https://indico.cern.ch/event/592612/contributions/2447142/ )
 affecting 2017, 2018 and 2023 scenarios in view of 900pre3

We suggested @ianna to re-do her talk at tomorrow's orp

**Comment by abdoulline** [comment]
Giovanni, I hope Walter will submit all the (DIGI-RECO) "Plan 0" tags 
later today. All the conditions are made available today by noon as txt 
inputs for those.

**Comment by franzoni** [comment]
Thanks @abdoulline 
https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2801/1.html

**Comment by franzoni** [comment]
hello @kpedro88 @abdoulline 
please cherry-pick  from this branch
 https://github.com/kpedro88/cmssw/compare/Phase1-HE14...franzoni:gf_Phase1-HE14_from-CMSSW_9_0_X_2017-01-15-2300-BIS?expand=1 
one commit which incoroporates , on top of the 3 geometry-related new tags of last week,
23 tags for the 2017 scenario ( https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2801.html ) where all the MC digitization and reconstruction -related tags
for HCAL
are modified to revert HE from phase1 to phase0

# With respect to the START of this PR, Summary of changes in Global Tags

## RunII simulation

   * **PhaseI realistic scenario** : [90X_upgrade2017_realistic_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_realistic_v1) as [90X_upgrade2017_realistic_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_realistic_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017_realistic_v1/90X_upgrade2017_realistic_v3):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0

   * **PhaseI design scenario** : [90X_upgrade2017_design_IdealBS_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_design_IdealBS_v1) as [90X_upgrade2017_design_IdealBS_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_design_IdealBS_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017_design_IdealBS_v1/90X_upgrade2017_design_IdealBS_v3):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0


   * **PhaseI cosmics scenario** : [90X_upgrade2017cosmics_realistic_peak_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v1) as [90X_upgrade2017cosmics_realistic_peak_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v1/90X_upgrade2017cosmics_realistic_peak_v3):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0

cheers,
Arun and Giovanni

**Comment by kpedro88** [comment]
don't test this yet, something isn't working

**Comment by abdoulline** [comment]
@franzoni    
Giovanni, GTs need to be remade, unfortunately...
There was a mistake when Walter was producing first bunch of tags yesterday.
Last night 17 of them have been re-made to v.2.0  
https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2802/1.html

Sorry for inconvenience.

**Comment by franzoni** [comment]
hello @kpedro88 @abdoulline 
please cherry-pick  from the same  branch
 https://github.com/kpedro88/cmssw/compare/Phase1-HE14...franzoni:gf_Phase1-HE14_from-CMSSW_9_0_X_2017-01-15-2300-BIS?expand=1 
the last commit.
It incorporates, top of what we released yesterday,
the 17 HCAL digi-reco updated tags which @walterjr released overnight (see: https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2802/1.html ) for the 2017 scenario w/ HE rolled back to phase0.
We also update the 3 GT's of the 2018 scenario, including tags properly named which contain the identical HCAL digi-reco payloads as made available in 81x for the full phase1 scenario.

# With respect to the START of this PR, Summary of changes in Global Tags

## RunII simulation

   * **PhaseI 2017 realistic scenario** : [90X_upgrade2017_realistic_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_realistic_v1) as [90X_upgrade2017_realistic_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_realistic_v4) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017_realistic_v1/90X_upgrade2017_realistic_v4):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0

   * **PhaseI 2017 design scenario** : [90X_upgrade2017_design_IdealBS_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_design_IdealBS_v1) as [90X_upgrade2017_design_IdealBS_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017_design_IdealBS_v4) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017_design_IdealBS_v1/90X_upgrade2017_design_IdealBS_v4):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0


   * **PhaseI 2017 cosmics scenario** : [90X_upgrade2017cosmics_realistic_peak_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v1) as [90X_upgrade2017cosmics_realistic_peak_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v4) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_upgrade2017cosmics_realistic_peak_v1/90X_upgrade2017cosmics_realistic_peak_v4):
      * HE geometry reverted back  to phase0
      * HCAL 23 digi-reco MC tags all modified to revert HE back to phase0


   * **PhaseI 2018 realistic scenario** : [90X_upgrade2018_realistic_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2018_realistic_v2) 
      * newly introduced
      * same content as the full phaseI detector, tags only renamed "2018"

   * **PhaseI 2018 design scenario** : [90X_upgrade2018_design_IdealBS_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2018_design_IdealBS_v2) 
      * newly introduced
      * same content as the full phaseI detector, tags only renamed "2018


   * **PhaseI 2018 cosmics scenario** : [90X_upgrade2018cosmics_realistic_peak_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/90X_upgrade2018cosmics_realistic_peak_v2) 
      * newly introduced
      * same content as the full phaseI detector, tags only renamed "2018

cheers,
Arun and Giovanni

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@slava77 these comparisons should be worth inspecting. I see expected features, e.g. no more SiPM dark current noise:
log10(HBHE energy)

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
unhold

**Comment by davidlange6** [comment]
hi all- please try to complete your review of this PR today. Thanks

**Comment by franzoni** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
Looking at jenkins 10024:

probably most notable is the reduction in the number of hits in HE and some "normalization" of chi2

all_oldvsnew_ttbar13tev2017wf10024p0c_log10hbherechitssorted_hbhereco_reco_obj_obj_chi220

while HB is roughly unchanged
all_oldvsnew_ttbar13tev2017wf10024p0c_log10hbherechitssorted_hbhereco_reco_obj_obj_chi210

population of calo towers looks almost unchanged
all_oldvsnew_ttbar13tev2017wf10024p0c_calotowerssorted_towermaker_reco_obj_obj_eta144
so, to first order, the response is OK.

simhit validation looks broken
wf10024_globalhits_hcal_eta
wf10024_globalhits_hcal_tof2


DigiPhase1Task has quite a few empty plots (probably OK), but some are filled
wf10024_digiphase1_hcal_depth4
is depth=4 here in the central from HO or something?

**Comment by slava77** [comment]
Some more plots, these are from the pi 5 to 5000 GeV gun

pf neutral hadrons show some changes in shape
all_sign831vsorig_pie5to5000in2017c_log10recopfcandidates_particleflow__reco_obj_pt73
no dramatic change, some variations at low pt, a fraction of this is probably noise or over-measured hits.

this matches cluster energy distribution
all_sign831vsorig_pie5to5000in2017c_log10recopfclusters_particleflowclusterhcal__reco_obj_energy

"jet" response looks OK as is
wfpi5to5000_ak4calo_gen200to600_eta

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17198	aae9443
- changes in reco code are rather minimal, just changing the era name
- jenkins tests pass and comparisons show differences only in 2017 workflows
- a local test with a pi gun in 2017 setup (derived from 10004.0) shows fairly consistent behavior in CMSSW_9_0_X_2017-01-27-1100

There is probably still something to fix in HCAL-specific DQM, but that appears to be in less critical plots (see examples in https://github.com/cms-sw/cmssw/pull/17198#issuecomment-275826503

**Comment by kpedro88** [comment]
@deguio @vkhristenko please check for any changes needed to accept TestNumbering-style SimHits in the HCAL DQM (already propagated through the validation code)

**Comment by vkhristenko** [comment]
HCAL DQM does not deal with sim hits.

**Comment by kpedro88** [comment]
Ah, I see: GlobalHitsV is a non-HCAL-specific validation module that looks at HCAL SimHits. @kencall can you look at it?

**Comment by hatakeyamak** [comment]
Is it possible to give a little more specific pointer so that it's easier for us to see which code/lines needs to be updated?

**Comment by kpedro88** [comment]
There are two DQM plots with "GlobalHitsV" in the name from one of Slava's comments https://github.com/cms-sw/cmssw/pull/17198#issuecomment-275826503. It seems like these must be produced somewhere in https://github.com/cms-sw/cmssw/tree/CMSSW_9_0_X/Validation/GlobalHits. That's about as far as my knowledge extends.

**Comment by hatakeyamak** [comment]
Okay. Thanks. I think we have to look around Validation/GlobalHits/src/GlobalHitsAnalyzer.cc. This is outside the HCAL validation code, and I don't know who uses these and who maintains. I see that most recent updates are done by wmtan <wmtan@fnal.gov>. Can somebody help figure out who is maintaining this package?

**Comment by slava77** [comment]
It looks like I missed warnings like
``` 
%MSG-w GlobalHitsAnalyzer_fillHCal:  GlobalHitsAnalyzer:globalhitsanalyze 
Unable to get CaloCellGeometry from HCalContainer for Hit 245
%MSG-w GlobalHitsAnalyzer_fillHCal:  GlobalHitsAnalyzer:globalhitsanalyze  28-Jan-2017 00:49:07 CET Run: 1 Event: 12
HCal PCaloHit 147 is expected to be (det,subdet) = (4,1 || 2 || 3 || 4); value returned is: (4,6)

```  which are happening 3.8K/event in my test pion gun sample.

**Comment by kpedro88** [comment]
Yes, that looks like the same thing I was asking Ken about. @hatakeyamak, can you or @kencall try to push TestNumbering-awareness into GlobalHits soon?

**Comment by sextonkennedy** [comment]
Hey @kpedro88 the XEB is asking about the status of this work.  What would you say is the probability of converging on a release at the ORP tomorrow?

**Comment by kpedro88** [comment]
@sextonkennedy my understanding had been that we had fixed all the handling of SimHit TestNumbering in the validation code. Apparently there is code outside of the HCAL domain that looks at SimHits. I don't know if a real fix will be ready tomorrow, but maybe a fake fix (i.e. suppressing the warning) could be sufficient.

For the record, here are all the warnings I see:
```
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/TimingCut/FED/ 
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/TimingCut/Electronics/ 
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/Occupancy/FED/ 
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/Occupancy/Electronics/ 
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/OccupancyCut/FED/ 
*** MonitorElement: WARNING:setBinLabel: attempting to set label of non-existent bin number for ME: Hcal/DigiPhase1Task/OccupancyCut/Electronics/ 
(repeated for DigiTask and RecHitTask)
```
```
%MSG-w GlobalHitsAnalyzer_fillHCal:  GlobalHitsAnalyzer:globalhitsanalyze  30-Jan-2017 18:11:31 CST Run: 1 Event: 1
HCal PCaloHit 1 is expected to be (det,subdet) = (4,1 || 2 || 3 || 4); value returned is: (1,0)
%MSG
%MSG-w GlobalHitsAnalyzer_fillHCal:  GlobalHitsAnalyzer:globalhitsanalyze  30-Jan-2017 17:00:34 CET Run: 1 Event: 101
Unable to get CaloCellGeometry from HCalContainer for Hit 46
%MSG
```
```
%MSG-w HcalDetId:  HcalDigisValidation:AllHcalDigisValidation  30-Jan-2017 18:11:31 CST Run: 1 Event: 1
HcalDetID(SimHit) presents conflicting information. Depth: 4, iphi: 34, ieta: 16. Max depth from geometry is: 2. TestNumber = 1
%MSG
```

@bsunanda @hatakeyamak @kencall

**Comment by slava77** [comment]
with git there should be no boundaries for HCAL-related developments to be easily applied to all packages in CMSSW.
The nominal strategy is that if you break the interface, you make best effort to change it throughout the release.

**Comment by davidlange6** [comment]
just to clarify, these errors occur even with fresh gen-sim

**Comment by kpedro88** [comment]
Yes, the warnings occur all the time, but the crashes observed in IB relvals do not.

Unfortunately git does not automatically grant us understanding of code we haven't looked at before. We will try to solve whatever can be solved today.

**Pull request merged**